### PR TITLE
feat: map for selecting location

### LINKED ISSUE #5: Map for selecting location
Instead of just typing the location in, display a map to the user (using OpenStreetMap for example), and take the coordinates of their pin and use that to display weather information.

Make it a 2 way thing, and if a user types a location in, it displays a pin on the map as well!

If you have an API key, let @wrussell1999 know so it can be added to the Netlify website hosting it!

### PR DESCRIPTION
**Pull request by Pravallika-Myneni**
#### Map of  user's location
image

image

#### When user types the location
image

closes #5

### CODE REVIEW

**Comment by joshi-kaushal** [comment]
@sumana2001 could you please help me here?

File: src/App.js [hunk 1]
```diff
@@ -10,14 +10,44 @@ function App() {
...
+              setCity(data[0].name);
+            })
+            .catch((error) => {
+              console.error(
+                `Couldn't fetch city based on user's location: ${error}`
+              );
+            });
+        },
+        (error) => {
+          console.error(
```
Comment (sahibardj) [inline src/App.js:45]: image
It is showing error even when the API is fetching the data.

Comment (sahibardj) [reply src/App.js:45]: Some issue is there with fetching coordinates of location look into that

**Comment by sahibardj** [comment]
If you using some API, please get it added in the main code as alignment of content is getting disrupted and map is not visible.

File: src/App.js [hunk 2]
```diff
@@ -25,71 +63,116 @@ function App() {
       .then((res) => res.json())
       .then(
         (result) => {
-            if (result["cod"] !== '200') {
+          if (result["cod"] !== "200") {
             setIsLoaded(false);
-            if(result["cod"]=="404")
-            {
+            if (result["cod"] == "404") {
```
Comment (sahibardj) [inline src/App.js:68]: Changes in code (possibly due to not fetching coordinates is setting notFound as true and  Data not found error text being printed on screen.
image

File: src/Search.js [hunk 3]
```diff
@@ -12,8 +12,8 @@ function autoCompleteCity(city) {
   return fetch(`${autocompleteURL}${query}`)
     .then((res) => res.json())
     .then((result) => {
-      console.log(result);
-      if(!result.items) return []
+      // console.log(result);
```
Comment (sahibardj) [inline src/Search.js:15]: Instead of commenting you can remove all the console.log statement

**Comment by sahibardj** [comment]
-Make the components responsive
image
.

**Comment by sahibardj** [comment]
image

-Make the component responsive (add margin to it so it looks better)

**Comment by sumana2001** [comment]
@Pravallika-Myneni @joshi-kaushal Please go through Sahiba's review and make the necessary changes so that we can merge this PR quickly

File: src/App.js (hunk 2 above)
Comment (joshi-kaushal) [reply src/App.js:68]: We didn't write this code, hence I have no clarity about this one.

**Comment by joshi-kaushal** [comment]
No, we are not using any external API for this. 
(https://github.com/MLH-Fellowship/prep-project-22.JUL.PREP.2/pull/27#issuecomment-1195181148)

File: src/Search.js (hunk 3 above)
Comment (joshi-kaushal) [reply src/Search.js:15]: solved.

**Comment by joshi-kaushal** [comment]
@sahibardj could you please review it again? We have made possible changes.

BTW, before submitting a review, please select **approve** instead of **comment**. Like this👇
image

**Review by sahibardj — APPROVED** [review]
Everything looks good.

**Comment by sahibardj** [comment]
Work on making it responsive as well.
image
image

**Comment by joshi-kaushal** [comment]
image
It looks responsive at my end. can you specify the range so i can look? Also, let's focus on standard sizes (laptop, tablet, mobile) please.

**Comment by joshi-kaushal** [comment]
@sumana2001 could you please help me out here? I have never seen this before
image

**Comment by sahibardj** [comment]
Map is not visible on my end (https://deploy-preview-27--prep-22-jul-prep-2-project.netlify.app/)
And even It is not responsive, I opened it in my mobile. I'm taking specifically about it not having margin with the right edge.

image

**Comment by joshi-kaushal** [comment]
When you open the app for the first time, it asks for your location and
renders the map. Please try from the incognito window and tell me if you
still have same problems.

**Comment by joshi-kaushal** [comment]
@sumana2001 everything is clear from our side, please have a look at yourself!

Also I suggest creating a PR for #34 since the current PR is already huge!

**Comment by sahibardj** [comment]
image
 
please check the deployment in your phone once, map is visible for me  but the issue with responsiveness is there.

**Comment by joshi-kaushal** [comment]
It was some CSS issue from forecast cards, fixed it. I tried to improve margins for the forecast as well but someone has added fixed margins to it hence it's pretty challenging to work with it.

**Review by sahibardj — APPROVED** [review]

**Comment by sahibardj** [comment]
@sumana2001 Please merge the PR.

**Pull request merged**